logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -111,7 +70,59 @@ def config_name
   end
 
   def register
-    @workers.each {|w| w.register}
+    raise ArgumentError, "Attempted to register #{self} twice!" if @registered
+    @registered = true
+    # We define this as an array regardless of threadsafety
+    # to make reporting simpler, even though a threadsafe plugin will just have
+    # a single instance
+    #
+    # Older plugins invoke the instance method Outputs::Base#workers_not_supported
+    # To detect these we need an instance to be created first :()
+    # TODO: In the next major version after 2.x remove support for this
+    @workers << @klass.new(@config)
+    @workers.first.register # Needed in case register calls `workers_not_supported`
```

Comment:
This is kind of crappy, but its required for backwards compat. The only way we know if a worker supports workers or not is if its register method calls 'workers_not_supported'. So, to determine if we should make multiple workers we need to: 

1. make a single worker and invoke register
1. Check if it invoked `workers_not_supported`
1. Start additional workers if so.

It sucks, and we should deprecate it in favore of [`declare_workers_supported!`](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/outputs/base.rb#L39) which is done at the class level